client/my-sites/plugins/access-control.js

```diff
@@ -66,6 +66,18 @@ const hasRestrictedAccess = ( site ) => {
 
 	site = site || sites.getSelectedSite();
 
+	const wpcomPluginPageError = {
```

Comment:
I very much like this improvement to my original flow, and I feel bad nitpicking this since my original code is so much worse... but this variable is used in two very specific cases in this rather complex function. Maybe it'd be better to extract the creation of this error object to a separate function and call it directly in those two cases?